Common/Python/PandasConverter.cs

```diff
@@ -226,5 +213,74 @@ private PyObject MakeIndicatorDataFrame(PyDict pyDict)
         {
             return _pandas.DataFrame(pyDict, columns: pyDict.Keys().Select(x => x.As<string>().ToLowerInvariant()).OrderBy(x => x));
         }
+
+        /// <summary>
+        /// Gets the <see cref="PandasData"/> for the given symbol if it exists in the dictionary, otherwise it creates a new instance with the
+        /// given base data and adds it to the dictionary
+        /// </summary>
+        private PandasData GetPandasDataValue(IDictionary<Symbol, PandasData> sliceDataDict, Symbol symbol, object data, ref int maxLevels)
+        {
+            PandasData value;
+            if (!sliceDataDict.TryGetValue(symbol, out value))
+            {
+                sliceDataDict.Add(symbol, value = new PandasData(data));
+                maxLevels = Math.Max(maxLevels, value.Levels);
+            }
+
+            return value;
+        }
+
+        /// <summary>
+        /// Adds each slice data to the pandas data dictionary
+        /// </summary>
+        private void AddSliceDataToDict(Slice slice, IDictionary<Symbol, PandasData> sliceDataDict, ref int maxLevels)
+        {
+            foreach (var key in slice.Keys)
+            {
+                var baseData = slice[key];
```

Comment:
Instead of iterating the keys and later accessing the value, can directly iterate the slice to get the kvp pair? Same in the method bellow

Reply:
We cannot iterate using the kvp, because it will enumerate on each base data instead of each symbol. For instance, if the slice has 3 ticks for a given symbol, enumerating the slice will result on three iterations where the symbol will be the same but the value will be each tick